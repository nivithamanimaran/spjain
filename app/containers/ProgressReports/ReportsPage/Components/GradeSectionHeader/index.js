import React, { useState } from 'react';
import { Grid, Typography, useTheme } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { useTranslation } from 'react-i18next';
import KenPopover from '../../../../../components/KenPopover/index';
import GradeIcon from '../GradeIcon/index';

const useStyles = makeStyles(theme => ({
  root: {
    height: 56,
    justifyContent: 'space-between',
    padding: 16,
  },
  primaryLabelColor: {
    color: theme.palette.KenColors.gradeSectionHeaderLabel,
    fontSize: 16,
  },
  secondaryLabelColor: {
    color: theme.palette.KenColors.gradeSectionHeaderLabel,
    cursor: 'pointer',
    fontSize: 13,
  },
  //
  gradeWrap: {
    background: theme.palette.KenColors.neutral11,
    borderRadius: 6,
    // border: `1px solid ${theme.palette.KenColors.gradeSectionBorder}`,
    boxShadow: `0px 0px 9px ${theme.palette.KenColors.shadowColor}`,
  },

  gradeLabelColor: {
    color: theme.palette.KenColors.neutral100,
    fontSize: 14,
    fontWeight: 600,
    display: 'flex',
    alignItems: 'center',
    padding: 16,
    background: theme.palette.KenColors.kenWhite,
  },
  gradeDetailsWrap: {
    padding: '8px 16px 16px',
    background: theme.palette.KenColors.kenWhite,
  },

  gradeMarksWrap: {
    display: 'flex',
    alignItems: 'center',
    padding: 8,
    whiteSpace: 'nowrap',
  },

  gradeValueWrap: {
    minWidth: 32,
    minHeight: 32,
    fontSize: 14,
    textAlign: 'center',
    fontWeight: 600,
    borderRadius: '50%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  gradeMarkLabel: {
    fontSize: 13,
    marginLeft: 7,
    color: theme.palette.KenColors.tertiaryBlue502,
  },
  pointLabel: {
    fontSize: 13,
    marginLeft: 32,
    color: theme.palette.KenColors.neutral400,
    fontWeight: 600,
  },
  achievementLabel: {
    fontSize: 13,
    marginLeft: 28,
    color: theme.palette.KenColors.neutral400,
  },
  remarksLabel: {
    color: theme.palette.KenColors.tertiaryRed503,
    fontSize: 12,
  },
}));

export default function GradeSectionHeader(props) {
  const classes = useStyles();
  const theme = useTheme();
  const {
    title,
    gradeLabel,
    gradeValue,
    gradePopupTitle,
    gradeDistribution,
    rowsPerGrid,
    popoverWidth,
    gradeColorPalette,
  } = props;
  const { t } = useTranslation();

  const [anchorEl, setAnchorEl] = React.useState(null);
  const open = Boolean(anchorEl);
  const divRef = React.useRef();

  const onPopupShow = () => {
    setAnchorEl(divRef.current);
  };

  const getGradeColor = grade => {
    return gradeColorPalette?.find(
      palette =>
        palette?.grade?.trim()?.toUpperCase() === grade?.trim()?.toUpperCase()
    );
  };

  return (
    <>
      <Grid
        container
        className={classes.root}
        data-testid="grade-section-header"
      >
        <Typography
          className={classes.primaryLabelColor}
          data-testid="grade-section-title"
        >
          {title}
        </Typography>
        <Typography
          className={classes.secondaryLabelColor}
          onClick={onPopupShow}
          data-testid="grade-section-label"
          ref={divRef}
        >
          {gradeLabel}
        </Typography>
      </Grid>
      <KenPopover
        open={anchorEl}
        anchorEl={anchorEl}
        handleClose={() => {
          setAnchorEl(null);
        }}
      >
        <Grid className={classes.gradeWrap} style={{ width: popoverWidth }}>
          <Typography className={classes.gradeLabelColor}>
            {gradePopupTitle}
          </Typography>
          <Grid container className={classes.gradeDetailsWrap}>
            {gradeDistribution?.map((item, index) => {
              return (
                <Grid
                  item
                  key={index}
                  md={rowsPerGrid}
                  xs={12}
                  className={classes.gradeMarksWrap}
                >
                  <Typography className={classes.gradeValueWrap}>
                    <GradeIcon
                      grade={item?.Grade}
                      {...getGradeColor(item.Grade)}
                    />
                  </Typography>
                  <Typography className={classes.gradeMarkLabel}>
                    {item?.marks}
                    <Typography className={classes.remarksLabel}>
                      {item?.remarks}
                    </Typography>
                  </Typography>

                  <Typography className={classes.pointLabel}>
                    {item?.points}
                  </Typography>
                  <Typography className={classes.achievementLabel}>
                    {/* <span>-</span> */}
                    {item?.achievement}
                  </Typography>
                </Grid>
              );
            })}
          </Grid>
        </Grid>
      </KenPopover>
    </>
  );
}
