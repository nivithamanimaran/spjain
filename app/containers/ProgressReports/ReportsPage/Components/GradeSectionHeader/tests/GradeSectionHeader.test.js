import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import ReactDOM from 'react-dom';
import GradeSectionHeader from '../index';
import ThemeProvider from '../../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

describe('Grade-Section-Header', () => {
  const div = document.createElement('div');
  it('Grade Section Header renders correctly without props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <GradeSectionHeader />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <GradeSectionHeader />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('grade-section-header')).toMatchSnapshot();
  });

  it('Grade Section Header renders correctly with props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <GradeSectionHeader title="title" gradeLabel="gradeLabel" />
        </Router>
      </ThemeProvider>,
      div
    );

    const title = queryByTestId('grade-section-title');
    expect(title).toBeTruthy();
    expect(title.textContent).toBe('Title');

    const gradeLabel = queryByTestId('grade-section-label');
    expect(gradeLabel).toBeTruthy();
    expect(gradeLabel.textContent).toBe('GradeLabel');
  });
});
