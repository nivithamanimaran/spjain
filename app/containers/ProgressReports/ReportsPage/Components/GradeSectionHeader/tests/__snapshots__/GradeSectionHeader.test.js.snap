// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grade-Section-Header Grade Section Header renders correctly without props 1`] = `
<div
  class="MuiGrid-root makeStyles-root-1 MuiGrid-container"
  data-testid="grade-section-header"
>
  <p
    class="MuiTypography-root makeStyles-primaryLabelColor-2 MuiTypography-body1"
    data-testid="grade-section-title"
  />
  <p
    class="MuiTypography-root makeStyles-secondaryLabelColor-3 MuiTypography-body1"
    data-testid="grade-section-label"
  />
</div>
`;
